src/date_type.h

```diff
@@ -7,17 +7,14 @@
 
 /** @file date_type.h Types related to the dates in OpenTTD. */
 
+#include "timer/timer_game_calendar.h"
```

Comment:
I don't expect this include here. I would expect all soruce-files to include `timer_game_calendar.h` when they need access to Year/Month/Day from calendar. That way, if we add economy, it is also more clear which source files make use of what timers.

I know, a lot of work to get the includes right .. been there, got the t-shirt. But it does make more sense :)